category/execution/monad/staking/fuzzer/staking_contract_model.cpp

```diff
@@ -0,0 +1,670 @@
+// Copyright (C) 2025 Category Labs, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include <category/execution/ethereum/core/contract/abi_encode.hpp>
+#include <category/execution/ethereum/core/contract/abi_signatures.hpp>
+#include <category/execution/monad/staking/fuzzer/staking_contract_model.hpp>
+
+namespace
+{
+    using namespace monad;
+
+    Result<u64_be> decode_u64_be_result(Result<byte_string> &&res)
+    {
+        BOOST_OUTCOME_TRY(auto const output, std::move(res));
+        MONAD_ASSERT(output.size() == 32);
+        u64_be x;
+        std::memcpy(x.bytes, output.data() + 24, 8);
+        return x;
+    }
+
+    Result<void> decode_true_result(Result<byte_string> &&res)
+    {
+        BOOST_OUTCOME_TRY(auto const output, std::move(res));
+        MONAD_ASSERT(output.size() == 32);
+        MONAD_ASSERT(output[31] == 1);
+        return outcome::success();
+    }
+}
+
+namespace monad::staking::test
+{
+    StakingContractModel::StakingContractModel()
+    {
+        trie_db_.commit(
```

Comment:
Why not initialize in the exact same way as `SetUp` in `test_staking_contract.cpp`?